website/gen_photo.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+"""
+Creates photometry table
+
+Attributes
+----------
+KNOWN_MAGS
+    Dictionary of photometric bands against central wavelength in microns
+
+Methods
+-------
+photometry_exists
+    Grabs photometry from object inventory on query
+main
+    Queries database for given object and finds photometry if it exists
+"""
+# local files
+from web_base import load_db, target_exists, target_argparse
+
+# global vars
+# TODO: expand known mags or use different method of initialising tables depending on if plotting photometry
+KNOWN_MAGS = {'WISE_W1': 3.368, 'WISE_W2': 4.618, 'WISE_W3': 12.082, 'WISE_W4': 22.194}  # central wave in microns
+
+
+def photometry_exists(inventory: dict):
+    """
+    Unpacks the photometry for an object
+
+    Parameters
+    ----------
+    inventory
+        Dictionary of the database output for that object
+    """
+    try:
+        if 'Photometry' not in inventory.keys():
+            raise NotImplementedError('No "Photometry" key yet in db')
+        elif type(inventory['Photometry']) != list:
+            raise TypeError('Photometry entry needs to produce list')
+    finally:
+        photo = inventory['Photometry']
```

Comment:
This line throws a KeyError even though you caught it above.

Reply:
Replaced ```finally``` with an ```except; else```